fn/req.go

```diff
@@ -0,0 +1,31 @@
+package fn
+
+// Req is a type to encapsulate RPC-like calls wherein we send some data
+// structure as a request, as well as a channel to receive the response on where
+// the remote goroutine will send the result.
+type Req[Input any, Output any] struct {
+	// request is the data we are sending to the remote goroutine for
+	// processing.
+	request Input
```

Comment:
Should these both be public? That would let `New` just have a single return value (ignoring the send/recv chan type for a min).

Reply:
Yes. They absolutely should. My mistake.